nixos/modules/config/i18n.nix

```diff
@@ -118,6 +118,29 @@
   ###### implementation
 
   config = {
+    warnings =
+      lib.optional
+        (
+          (lib.subtractLists config.i18n.supportedLocales (
+            builtins.map
+              (l: (lib.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] l) + "/UTF-8")
+              (
+                [ config.i18n.defaultLocale ]
+                ++ config.i18n.extraLocales
+                ++ (lib.attrValues (lib.filterAttrs (n: v: n != "LANGUAGE") config.i18n.extraLocaleSettings))
+              )
+          )) != [ ]
```

Comment:
BTW I'm sorry to be so demanding, but: Do you think you can define this function externally? It is very repetitive. Also, perhaps the `supportedLocales.defaultText` that also copy pastes this logic almost precisely, can be removed since this option is now hidden? What do you think? It could be a great improvement I think.

Reply:
Ok, refactored it into an aggregated variable.

I removed the default value description, but kept the other description and example because I think it should still be useful for future maintainers.